### PR TITLE
Add interactivity docs

### PR DESCRIPTION
**Pull request by davidmanzanares**
Fixes https://github.com/CartoDB/renderer-prototype/issues/227 and https://github.com/CartoDB/renderer-prototype/issues/201

### CODE REVIEW

File: src/api/interactivity.js [hunk 1]
```diff
@@ -3,6 +3,66 @@ import Layer from './layer';
 import { WM_R, projectToWebMercator } from './util';
 import { wToR } from '../client/rsys';
 
+
+/**
+ * 
+ * FeatureEvent objects are fired by {@link carto.Interactivity|Interactivity} objects.
```
Comment (IagoLast) [inline src/api/interactivity.js]: what is a feature event? what is it used for?

File: src/api/interactivity.js [hunk 2]
```diff
@@ -3,6 +3,66 @@ import Layer from './layer';
...
+ * @typedef {Object} FeatureEvent
+ * @property {object} coordinates LongLat coordinates in {lng: 0, lat:0} form
```
Comment (IagoLast) [inline src/api/interactivity.js]: missing id property

File: src/api/interactivity.js [hunk 3]
```diff
@@ -3,6 +3,66 @@ import Layer from './layer';
...
+
+
+/**
+ * featureClick events are fired when the user clicks on features. The list of features behind the cursor is provided.
+ *
+ * @event featureClick
+ * @type {FeatureEvent}
+ * @api
+ */
+/**
```
Comment (IagoLast) [inline src/api/interactivity.js]: add white line to split

File: src/api/interactivity.js [hunk 4]
```diff
@@ -11,29 +71,68 @@ const EVENTS = [
...
+    *
+    * Interactivity purpose is to allow the reception and management of user-generated events, like clicking, over layer features.
+    *
+    * To create a Interactivity object an array of {@link carto.Layer} is required.
+    * Events fired from interactivity objects will refer to the features of these layers and only these layers.
+    *
+    * @param {Array<carto.Layer>} layerList - Array of {@link carto.Layer}, events will be fired based on the features of these layers. The array cannot be empty, and all the layers must be attached to the same map.
+    *
+    * @example
+    * const myLayer = new carto.Layer('layer0', mySource, myStyle);
```
Comment (IagoLast) [inline src/api/interactivity.js]: this example is wrong, we must wait for the . `layer.loaded` event to create the interactivity.

In the future, I think we should make the addtoMap async

File: src/api/interactivity.js [hunk 5]
```diff
@@ -11,29 +71,68 @@ const EVENTS = [
...
         this._init(layerList);
-        
-        this._prevFeatures = [];
-    }
 
-    _init(layerList) {
-        this._emitter = mitt();
-        this._layerList = layerList;
-        this._subscribeToIntegratorEvents(layerList[0].getIntegrator());
+        this._prevFeatures = [];
```
Comment (IagoLast) [inline src/api/interactivity.js]: this should be moved to the `init` method

File: src/api/interactivity.js (hunk 1 above)
Comment (davidmanzanares) [reply src/api/interactivity.js]: It is an `Object`, usage is end-app scope, not ours.

File: src/api/interactivity.js (hunk 2 above)
Comment (davidmanzanares) [reply src/api/interactivity.js]: Missing on `Feature`, not here, and that's not implemented on `master`

File: src/api/interactivity.js (hunk 4 above)
Comment (davidmanzanares) [reply src/api/interactivity.js]: Yes

File: src/api/interactivity.js (hunk 5 above)
Comment (davidmanzanares) [reply src/api/interactivity.js]: This is certainly inconsistent (although it was there before this PR), however, I doubt we need a `_init` method in the first place, it is pure side-effect and it is only called by the constructor, also, it is 4 lines long

File: src/api/interactivity.js (hunk 4 above)
Comment (davidmanzanares) [reply src/api/interactivity.js]: Fixed

**Pull request merged**